components/raftstore/src/store/fsm/peer.rs

```diff
@@ -5897,8 +5915,14 @@ where
         }
         self.fsm.peer.size_diff_hint = 0;
         self.fsm.peer.compaction_declined_bytes = 0;
-        // the task is scheduled, next tick may skip it.
-        self.fsm.peer.may_skip_split_check = true;
+        // the task is scheduled, next tick may skip it only when the size and keys are
+        // small If either size or keys are big enough to do a split,
+        // keep split check tick until split is done
+        if !matches!(self.fsm.peer.may_split_size, Some(true))
+            && !matches!(self.fsm.peer.may_split_keys, Some(true))
+        {
+            self.fsm.peer.may_skip_split_check = true;
```

Comment:
`may_skip_split_check` complicates the mechanism which looks quite confusing

Reply:
It's necessary, otherwise too much cost on split check task.